pyiron_continuum/schroedinger/schroedinger.py

```diff
@@ -0,0 +1,268 @@
+from scipy.constants import physical_constants
+import matplotlib.pyplot as plt
+
+try:
+    import k3d
+    k3d_alarm = ImportAlarm()
+except ImportAlarm:
+    k3d_alarm = ImportAlarm(
+        '3d plotting requires the k3d package which is accessible by `conda install -c conda-forge k3d`. Using k3d in '
+        'a jupyter environment further requires the following shell commands: '
+        '`jupyter nbextension install --py --sys-prefix k3d; jupyter nbextension enable --py --sys-prefix k3d`.'
+    )
+
+
+# TODO: Convert to pyiron units
+HBAR = 1  # EV_TO_U_ANGSTROMSQ_PER_SSQ * physical_constants['Planck constant in eV/Hz'][0] / (2 * np.pi)
+M_EL = 1  # physical_constants['electron mass in u'][0]
+KB = physical_constants['Boltzmann constant in eV/K'][0]
+
+__author__ = "Liam Huber"
+__copyright__ = (
+    "Copyright 2021, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Computational Materials Design (CM) Department"
+)
+__version__ = "0.1"
+__maintainer__ = "Liam Huber"
+__email__ = "huber@mpie.de"
+__status__ = "development"
+__date__ = "Sep 8, 2021"
+
+
+class _TISEInput(DataContainer):
+    def __init__(self, init=None, table_name='input', lazy=False):
+        super().__init__(init=init, table_name=table_name, lazy=lazy)
+        self._mesh = None
+        self._potential = None
+        self.n_states = 1
+        self.mass = M_EL
+
+    @property
+    def mesh(self) -> RectMesh:
+        return self._mesh
+
+    @mesh.setter
+    def mesh(self, new_mesh: RectMesh):
+        if not isinstance(new_mesh, RectMesh):
+            raise TypeError(f'Meshes must be of type {RectMesh} but got type {type(new_mesh)}')
+        new_mesh.simplify_1d = False
+        self._mesh = new_mesh
+
+    @property
+    def potential(self):
+        return self._potential
+
+    @potential.setter
+    def potential(self, new_potential):
+        self._potential = new_potential
+
+
+class _TISEOutput(DataContainer):
+    def __init__(self, init=None, table_name='input', lazy=False):
+        super().__init__(init=init, table_name=table_name, lazy=lazy)
+        self.energy = None
+        self.psi = None
+        self._rho = None
+
+    @property
+    def _space_axes(self):
+        return tuple(n for n in range(1, len(self.psi.shape)))
+
+    def _broadcast_vector_to_space(self, vector):
+        return vector.reshape(-1, *(1,)*len(self._space_axes))
+
+    def _normalize_per_state(self, states):
+        return states / self._broadcast_vector_to_space(np.linalg.norm(states, axis=self._space_axes))
+
+    def _weight_states(self, states, weights):
+        return states * self._broadcast_vector_to_space(weights)
+
+    @property
+    def rho(self):
+        if self._rho is None and self.psi is not None:
+            self._rho = self._normalize_per_state(self.psi ** 2)
+        return self._rho
+
+    def get_boltzmann_occupation(self, temperature):
+        if self.energy is not None:
+            return 1. / (np.exp(self.energy / (KB * temperature)) + 1)
```

Comment:
Is the `+ 1` term at the end necessary?

Reply:
Yes, it is necessary...for Fermi Dirac statistics. Somewhere in my development I was playing around with FD statistics and the function name and the function physics got mixed up!!! This is now properly the boltzmann weight, so please retry this code on your high-T example. 

We are absolutely allowed to construct the thermalized rho either by weighting individual state rho's or by weighting individual state psi's and then taking the |Psi|^2. But between my incorrect normalization for rho and my wrong statistics for the state weights we were getting garbage out of the code.

The lesson here is to always have unit tests... For `Potential` and `RectMesh` I was really developing tests either before (i.e. 'test-driven') or alongside the code development, then for `TISE` I was getting impatient. The -- PREDICTABLE! -- result is that I made mistakes. /wisdom